contracts/base/LockAndMsgSender.sol

```diff
@@ -2,19 +2,17 @@
 pragma solidity ^0.8.24;
 
 import {Constants} from '../libraries/Constants.sol';
+import {Locker} from '../libraries/Locker.sol';
 
 contract LockAndMsgSender {
```

Comment:
this contract seems a little silly lol

Reply:
It definitely is 😂 honestly i think the `map` function doesnt belong here... feels like the lock and the msg.sender logic should be separated

Reply:
ok i've moved `map` to the main dispatcher file because it doesnt feel like it belongs in the lock. And added a _msgSender function to make the lock msg.sender logic clearer

Reply:
great!